modules/ROOT/pages/api-gateway-encryption-mule4.adoc

```diff
@@ -0,0 +1,270 @@
+--
+
+== Passing your Encrypted Client ID and Secret
+
+After encrypting your credentials, you need to configure your encrypted values in your `wrapper.conf` file located inside the `conf` directory in the root folder of your Mule runtime:
+
+[source]
+--
+anypoint.platform.client_id.<n>
+anypoint.platform.client_secet.<n>
+anypoint.platform.proxy_password.<n>
+--
++
+[CAUTION]
+--
+If you add `wrapper.java.additional.n` entries to the configuration file, you must change each instance of `<n>` to a consecutive number, or Java does not parse the properties correctly.
+--
++
+See xref:mule-runtime::passing-additional-arguments-to-the-jvm-to-control-mule.adoc[Passing Additional Arguments to the JVM to Control Mule] for more information.
+
+In the example above, your `client_id` would look as below:
+
+[source]
+--
+anypoint.platform.client_id.<n>=![smzbg+aV2HbiC9Fv8KBsZpD5PlXiplsfhZ42RWNEbHhUeZ9ZIDftljUx6sYpuQ+b]
+--
+
+[NOTE]
+The `![` prefix and  `]` suffix are required, as they indicate the runtime that the enclosed value is encrypted.
+
+For the runtime to be able to decrypt your secured values, you need to pass your secret key when starting your Mule runtime using the `anypoint.platform.encryption_key` property:
+
+[source]
+--
+$MULE_HOME/bin/mule start -Danypoint.platform.encryption_key=MyEncryptionKey1
+--
+
+The runtime uses the encryption key passed in the `anypoint.platform.encryption_key` property to decrypt the properties.
+
+=== How It Works
+
+You can encrypt your Anypoint Platform credentials (client ID, client secret, and your proxy password) and add those encrypted values, along with your encryption key to your runtime's `wrapper.conf` file.
+
+While the runtime is starting it checks for an encryption key. Then it gets the client ID and client secret values from its `wrapper.conf` file. If some of the values in that file are enclosed within the encryption indicators (`![` and `]`), the runtime uses the encryption key it loaded earlier to decrypt the values and use its original content.
+
+== Debugging Startup Encrpytion
+
+You can configure your gateway logger to display information about the encryption mechanism during startup.
+
+In `$MULE_HOME/conf/log4j2.xml`, set up an asynchronous logger for your gateway deployment:
+
+[source,XML,linenums]
+--
+...
+  <AsyncLogger name="com.mulesoft.mule.runtime.gw.deployment" level="DEBUG" />
+...
+--
+
+When you provide a key, the runtime displays the following message:
+
+----
+An encryption key is present. Policies and API contracts will be encrypted.
+----
+
+If not, the runtime stars with:
+
+----
+No encryption key provided. Policies and API contracts won't be encrypted.
+----
+
+If the decryption process fails, the runtime displays an error:
+
+----
+Client ID cannot be decrypted
+----
+
+If your credentials can't be decrypted, you cannot establish a connection with Platform, and won't be able to download policies or contracts.
```

Comment:
what is "Platform" here? "Anypoint Platform" ? or may be just use "...with the Platform".

Reply:
Right: This should be Anypoint Platform.